astropy/io/fits/_tiled_compression/codecs.py

```diff
@@ -0,0 +1,362 @@
+"""
+This module contains the FITS compression algorithms in numcodecs style Codecs.
+"""
+from gzip import compress as gzip_compress
+from gzip import decompress as gzip_decompress
+
+import numpy as np
+
+from astropy.io.fits._tiled_compression._compression import (
+    compress_hcompress_1_c,
+    compress_plio_1_c,
+    compress_rice_1_c,
+    decompress_hcompress_1_c,
+    decompress_plio_1_c,
+    decompress_rice_1_c,
+)
+
+try:
+    from numcodecs.abc import Codec
+except ImportError:
+
+    class Codec:
+        codec_id = None
+
+
+__all__ = [
+    "Gzip1",
+    "Gzip2",
+    "Rice1",
+    "PLIO1",
+    "HCompress1",
+]
+
+
+class Gzip1(Codec):
+    """
+    The FITS GZIP 1 compression and decompression algorithm.
+
+    The Gzip algorithm is used in the free GNU software compression utility of
+    the same name. It was created by J. L. Gailly and M. Adler, based on the
+    DEFLATE algorithm (Deutsch 1996), which is a combination of LZ77 (Ziv &
+    Lempel 1977) and Huffman coding.
+    """
+
+    codec_id = "FITS_GZIP1"
+
+    def decode(self, buf):
+        """
+        Decompress buffer using the GZIP_1 algorithm.
+
+        Parameters
+        ----------
+        buf
+            The buffer to decompress.
+
+        Returns
+        -------
+        buf
+            The decompressed buffer.
+        """
+        cbytes = np.frombuffer(buf, dtype=np.uint8).tobytes()
+        dbytes = gzip_decompress(cbytes)
+        return np.frombuffer(dbytes, dtype=np.uint8)
+
+    def encode(self, buf):
+        """
+        Compress the data in the buffer using the GZIP_1 algorithm.
+
+        Parameters
+        ----------
+        buf
+            The buffer to compress.
+
+        Returns
+        -------
+        buf
+            A buffer with compressed data.
+        """
+        dbytes = np.asarray(buf).tobytes()
```

Comment:
Again, no need for `.tobytes()` (independent of `dtype`).

Reply:
Fixed